mava/components/tf/modules/exploration/exploration_scheduling.py

```diff
@@ -235,3 +236,69 @@ def decrement_epsilon(self) -> float:
             constant epsilon.
         """
         return self._epsilon
+
+
+class PieceWiseLinearTimestepScheduler:
+    """A piecewise linear scheduler for epsilon.
+
+    Args:
+        timesteps : list of timesteps at which epsilon is changed. The list should be in
+        ascending order.
+        epsilons : list of epsilon values at each timestep.
+        initial_fallback_epsilon : epsilon value to use when timestep is less than the
+        first timestep in the list.
+        final_fallback_epsilon : epsilon value to use when timestep is greater than the
```

Comment:
I don't think we should use `initial_fallback_epsilon` and `final_fallback_epsilon`. I think we should stick to an implementation close to https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/PiecewiseConstantDecay, where the first epsilon in the list (`epsilon[0]`) is used until the first `timesteps[0]`. 

It would be great to test that this and the tensorflow scheduler behave similiarily. That way we can use for our jax systems, without need to rely on tf :)

Reply:
Tensorflow has piecewise constant schedule, but for an RL project it makes more sense to have piecewise linear. Something similar to what is implemented in RLlib as in [here](https://docs.ray.io/en/latest/_modules/ray/rllib/utils/schedules/piecewise_schedule.html#PiecewiseSchedule).

That is a choice to be made, I guess it make more sense to have a piecewise linear exploration. Please let me know what you think.

Reply:
Thanks for clarifying @nima-siboni ! Makes more sense now. Could we please follow @sash-a feedback about `prepend [ing] initial_fallback_epsilon and final_fallback_epsilon to epsilons`. This would lead to less possible confusion, e.g. now when I init the component and then call `get_epsilon` , I expect to get `initial_fallback_epsilon` from the description , but instead I believe we get `epsilon[0]`.

Having an `epsilon` and `timesteps` would simplify this a lot and make it easier for people to use.